pkg/controllers/workgenerator/controller.go

```diff
@@ -850,6 +931,9 @@ func buildAllWorkAppliedCondition(works map[string]*fleetv1beta1.Work, binding *
 			ObservedGeneration: binding.GetGeneration(),
 		}
 	}
+	// In the case where ReportDiff apply strategy is used, work applier will no longer update
+	// Applied condition on Work objects; since existing Applied conditions will all become stale,
+	// this function will return a False Applied condition.
```

Comment:
do we still need this comment?

Reply:
Hi Ryan! If the rollout controller has been revised to always push changes to bindings under the ReportDiff mode (either by treating all such bindings as ready ones or failed ones), I think it might make sense to not set Applied condition at all in the work generator with the ReportDiff mode (which, with the current logic, would have the Applied condition removed upon reconciliation).

Reply:
Edited after the latest discussed change.